app/utils/importHelpers.ts

```diff
@@ -847,25 +881,25 @@ export async function importAddressBookEntries(
         }
         const sameName = duplicate.name === match.name;
         const sameNote = duplicate.note === match.note;
-        if (sameName && sameNote) {
+        const sameWriteStatus = duplicate.readOnly === match.readOnly;
+        if (sameName && sameNote && sameWriteStatus) {
             trueDuplicates.push(duplicate);
         } else {
             const update: Partial<AddressBookEntry> = {};
             if (!sameName) {
                 const {
                     chosenName,
                     message,
-                } = await resolveAddressBookNameConflict(
-                    match.name,
-                    duplicate.name,
-                    address
-                );
+                } = await resolveAddressBookNameConflict(match, duplicate);
                 update.name = chosenName;
-                addMessage(address, message);
+                addMessage(address, message, true);
```

Comment:
Try to avoid parameterizing functions with a boolean. When you read
```
addMessage(address, message, true);
```
it is hard to know what `true` indicates. It becomes much simpler if it is just:
```
addAddressBookMessage(address, message);
```

Reply:
I agree, I have refactored, let me know what you think.